flair/data.py

```diff
@@ -198,22 +200,76 @@ def __init__(self, text: str = None, use_tokenizer: bool = False, labels: List[L
 
         self._embeddings: Dict = {}
 
-        # optionally, directly instantiate with sentence tokens
+        # if text is passed, instantiate sentence with tokens (words)
         if text is not None:
 
-            # tokenize the text first if option selected, otherwise assumes whitespace tokenized text
+            # tokenize the text first if option selected
             if use_tokenizer:
-                sentences = split_single(text)
+
+                # use segtok for tokenization
                 tokens = []
+                sentences = split_single(text)
                 for sentence in sentences:
                     contractions = split_contractions(word_tokenizer(sentence))
                     tokens.extend(contractions)
 
-                text = ' '.join(tokens)
+                # determine offsets for whitespace_after field
+                index = text.index
+                running_offset = 0
+                last_word_offset = -2
+                last_token = None
+                for word in tokens:
+                    token = Token(word)
+                    self.add_token(token)
+                    try:
+                        word_offset = index(word, running_offset)
+                    except:
+                        word_offset = last_word_offset = +1
+                    if word_offset - 1 == last_word_offset:
+                        last_token.whitespace_after = False
+                    word_len = len(word)
+                    running_offset = word_offset + word_len
+                    last_word_offset = running_offset - 1
+                    last_token = token
+
+            # otherwise assumes whitespace tokenized text
+            else:
+                # add each word in tokenized string as Token object to Sentence
+                for word in text.split(' '):
+                    token = Token(word)
+                    self.add_token(token)
+
+    def _infer_space_after(self):
+        """
+        Heuristics in case you wish to infer whitespace_after values for tokenized text. This is useful for some old NLP
+        tasks (such as CoNLL-03 and CoNLL-2000) that provide only tokenized data with no info of original whitespacing.
+        :return:
+        """
+        last_token = None
+        quote_count: int = 0
+        # infer whitespace after field
+
+        for token in self.tokens:
+            if token.text == '"':
+                quote_count += 1
+                if quote_count % 2 != 0:
+                    token.whitespace_after = False
+                elif last_token is not None:
+                    last_token.whitespace_after = False
```

Comment:
This does not work for a sentence like: `xyz "abc"`. With the current code you would set `whitespace_after` to False for `xyz`.

Reply:
I think it would work: 

```python
sentence: Sentence = Sentence('xyz "abc".', use_tokenizer=True)
print(sentence.to_tokenized_string())
print(sentence.to_plain_string())
```
gives me: 
 
```
xyz " abc " .
xyz "abc".
```

whitespace_after here depends on the `quote_count`.